src/components/WeatherWidget.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <div class="weather-widget">
+    <WeatherWidgetPlace
+      :="weatherData.location"
+      :date="selectedDayData.date ?? 'Today'"
+    />
+    <WeatherWidgetCurrentData :="selectedDayData" />
+    <WeatherWidgetForecast
+      :forecastData="weatherData.forecast"
+      :selectedDay="selectedDay"
+      @handleDayChange="handleDayChange"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import WeatherWidgetPlace from "./WeatherWidgetPlace.vue";
+import WeatherWidgetCurrentData from "./WeatherWidgetCurrentData.vue";
+import WeatherWidgetForecast from "./WeatherWidgetForecast.vue";
+import { computed, ref } from "vue";
+import { mockWeatherResponseData } from "../helpers/mockWeatherDataResponse";
+import { mapWeatherData } from "../utils.ts/dataMappers";
+
+const selectedDay = ref<number | null>(null);
+const weatherData = mapWeatherData(mockWeatherResponseData);
```

Comment:
for now it works, but to make it reactive (fe after fetching new data) it needs to go to computed